### ISSUE TITLE
Issue reconnecting from sleep

### ISSUE THREAD

**Issue by Omnicrash** [samartzidis/magicstick.io#12]
Ever since one of the more recent updates, I've been having trouble reconnecting the keyboard when the PC has been in sleep.

- I put the PC to sleep
- I turn off the keyboard
- Later, I power the PC back on
- The stick shows a solid green LED, indicating a connection
- PC won't accept input until unplugging and plugging the stick

Strangely, I do think the keyboard can still wake the PC? (I need to verify this to know 100%, I'll keep an eye on it and update this issue).

**Comment by samartzidis** [comment]
Thanks for reporting this. It must be intermittent as I could not reproduce this but I'll take a closer look. Please update me on the issue if you find more accurate information on how to reproduce this consistently.

**Comment by Omnicrash** [comment]
It seems to be consistently happening now on my end.
- Put PC to sleep
- Wake it by pressing buttons on the keyboard (this works)
- Once awake, it requires a re-plug to accept inputs

What is the best way do do a full reset? I'm going to try that first to see if it makes a difference.

**Comment by samartzidis** [comment]
Ok thank you. I will review the relevant sleep-wakeup code. In the meantime
if you email me, I am happy to send you the previous firmware (the one
without the programming features). The reset procedure using the bootsel
button is described in the user manual.

**Comment by Omnicrash** [comment]
I found it happens only when waking the PC through the keyboard - if I press the power button on the PC to wake it from sleep, it will work fine.

Some more possibly relevant info for my system:
- Windows 11
- Stick is connected through an USB hub

**Comment by Omnicrash** [comment]
A full reset didn't fix it.

I've also noticed when this happens, operations from the UI like Settings or Keymap will result in a timeout error.

**Comment by samartzidis** [comment]
Hi @Omnicrash. As said earlier I am not able to reproduce this over Bluetooth or wired connection. I wonder what could be causing this in your particular setup. Are you running the currently latest firmware magicstick.2.0.155+release.6569e43 ?
Thanks.

**Comment by Omnicrash** [comment]
Yes it's on the latest version: [Device Model, magicstick.2.0.155+release.6569e43].
The keyboard itself should also be the latest magic keyboard (without fingerprint).
If there are specific things you'd like me to try to help you debug, let me know.

**Comment by Omnicrash** [comment]
I've been able to confirm that it works using the old firmware. I'll do some more testing with leaving the PC off for longer and waking it up, but I've tried:
- Putting it to sleep, then waking it up using the keyboard.
- Putting it to sleep, turning the keyboard off, turning it back on, then waking the PC.
Both do work so far, for the newer FW this would immediately fail.

Maybe another detail that might not matter, but just to rule out edge cases: The way I usually wake the PC is by mashing left ctrl and left shift, alternatingly or at the same time.

**Comment by samartzidis** [comment]
Thanks for testing @Omnicrash 

In the meantime, for flashing older firmware releases, can always visit:
https://github.com/samartzidis/magicstick.io/blob/main/release-notes.md

**Comment by samartzidis** [comment]
Even though not explicitly targeting your reported issue, as I could not reproduce it. There is a new [beta release](https://github.com/samartzidis/magicstick/releases/tag/magicstick.3.0.0-beta) that might have resolved it. Perhaps worth giving it a try? Thanks!

**Comment by Omnicrash** [comment]
Thanks, I'll try it out and report back.

**Comment by Omnicrash** [comment]
Unfortunately, the issue still persists.
Let me know if there's anything else I can test in the future.

**Comment by samartzidis** [comment]
OK thanks for the feedback. Unfortunately without being able to ever reproduce the issue it is very difficult to advise on a solution. I also have not heard of anyone else reporting this so must be something very specific to your setup/configuration. I'll keep this this open for now. Thanks!
